Add source-code editor and simple integration

smacker opened the pull request:
<img width="1279" alt="screen shot 2018-05-21 at 15 54 00" src="https://user-images.githubusercontent.com/406916/40311117-38bba204-5d0f-11e8-9148-37e12a75a45c.png">

HOC returns only props due need to support different layouts and it's an application responsibility, not a lib.

carlosms commented on src/withUASTEditor.js:
```diff
@@ -0,0 +1,115 @@
+        return;
+      }
+      const { uast, lastHighlighted } = this.state;
+      const newUast = highlightNodeById(uast, pos.id, lastHighlighted);
```
I think I've found a bug here. If you click on the same word, the node gets un-highlighted.

One possible solution:

```diff
 export function highlightNodeById(uast, id, prevId) {
   const node = uast[id];
   const newUast = {
     ...uast,
     [id]: {
       ...node,
       highlighted: true
     }
   };
-  if (prevId !== null) {
+  if (prevId !== null && prevId !== id) {
     newUast[prevId] = {
       ...newUast[prevId],
       highlighted: false
     };
   }

```

smacker replied:
fixed

carlosms commented on src/withUASTEditor.js:
```diff
@@ -0,0 +1,115 @@
+      };
+    }
+
+    onNodeHover(id, prevId) {
```
In the demo I'm seeing something strange.

The node hover effect is persistent, when the mouse exists the UAST tree you still get the highlighted text in the editor. But then you click in the text, and a different UAST node gets highlighted. I think there should be a way to clear state.pos when the mouse exists the UAST tree.

smacker replied:
fixed

carlosms commented on src/Editor.js:
```diff
@@ -0,0 +1,126 @@
+    const options = {
+      mode: languageMode,
+      lineNumbers: true,
+      theme: 'solarized light'
```
Maybe the theme should be open to be configured via props, with this one as default.

bzz commented:
@carlosms seems like feedback has been addressed and it's time for another pass.

@smacker seems like a rebase is in order here

smacker commented:
rebased. Commits are squashed to simplify rebase.

Demo moved to styleguide:
<img width="1304" alt="screen shot 2018-05-22 at 11 57 37" src="https://user-images.githubusercontent.com/406916/40355683-5fdddf8e-5db7-11e8-88e7-bf3c4d7a151c.png">

Added md file for Editor component.
